cpp/include/cuspatial/experimental/detail/point_linestring_distance.cuh

```diff
@@ -0,0 +1,152 @@
+
+  auto const idx = threadIdx.x + blockIdx.x * blockDim.x;
+  // Pointer to the last point in linestring array.
+  if (linestring_points_first + idx >= thrust::prev(linestring_points_last)) { return; }
+
+  auto offsets_iter =
+    thrust::upper_bound(thrust::seq, linestring_offsets_first, linestring_offsets_last, idx);
+  // Pointer to the last point in the linestring.
+  if (*offsets_iter - 1 == idx) { return; }
+
+  auto pair_idx = thrust::distance(linestring_offsets_first, thrust::prev(offsets_iter));
+  cartesian_2d<T> const a  = linestring_points_first[idx];
+  cartesian_2d<T> const b  = linestring_points_first[idx + 1];
+  cartesian_2d<T> const c  = points_first[pair_idx];
+
+  auto const distance_squared = point_to_segment_distance_squared(c, a, b);
+
+  atomicMin(&thrust::raw_reference_cast(*(distances + pair_idx)),
+            static_cast<T>(std::sqrt(distance_squared)));
+}
+
+}  // namespace detail
+
+template <class Cart2dItA, class Cart2dItB, class OffsetIterator, class OutputIt>
+void pairwise_point_linestring_distance(Cart2dItA points_first,
+                                        Cart2dItA points_last,
+                                        OffsetIterator linestring_offsets_first,
+                                        Cart2dItB linestring_points_first,
+                                        Cart2dItB linestring_points_last,
+                                        OutputIt distances_first,
+                                        rmm::cuda_stream_view stream)
+{
+  using T = typename std::iterator_traits<Cart2dItA>::value_type::value_type;
+
+  static_assert(
+    detail::is_floating_point<T,
+                              typename std::iterator_traits<Cart2dItB>::value_type::value_type,
+                              typename std::iterator_traits<OutputIt>::value_type>(),
+    "Inputs and output must have floating point value type.");
+
+  static_assert(detail::is_same<T,
+                                typename std::iterator_traits<Cart2dItB>::value_type::value_type,
+                                typename std::iterator_traits<OutputIt>::value_type>(),
+                "Inputs and output must have the same value type.");
+
+  static_assert(detail::is_same<cartesian_2d<T>,
+                                typename std::iterator_traits<Cart2dItA>::value_type,
+                                typename std::iterator_traits<Cart2dItB>::value_type>(),
+                "Inputs must be cuspatial::cartesian_2d");
+
+  auto const num_pairs = thrust::distance(points_first, points_last);
+
+  if (num_pairs == 0) { return; }
+
+  auto const num_linestring_points =
+    thrust::distance(linestring_points_first, linestring_points_last);
+
+  thrust::fill(rmm::exec_policy(stream),
+               distances_first,
+               distances_first + num_pairs,
+               std::numeric_limits<T>::max());
+
+  std::size_t constexpr threads_per_block = 64;
```

Comment:
Seems a very small block size. How did you choose this?

Reply:
Pretty much migrated from `hausdorff` kernel. Would it be nice to benchmark on the kernel and find out about the best tpb?

Reply:
Yes. Generally I start from 256. I probably wouldn't have said anything if this was 256.

Not sure why Hausdorff is only 64!

Reply:
https://github.com/rapidsai/cuspatial/issues/651